ldms/src/decomp/static/decomp_static.c

```diff
@@ -1030,6 +1030,18 @@ static void assign_value(ldms_mval_t dst, ldms_mval_t src,
 	case LDMS_V_CHAR_ARRAY:
 		memcpy(dst->a_char, src->a_char, count);
 		break;
+	case LDMS_V_U8_ARRAY:
+	case LDMS_V_S8_ARRAY:
+	case LDMS_V_U16_ARRAY:
+	case LDMS_V_S16_ARRAY:
+	case LDMS_V_U32_ARRAY:
+	case LDMS_V_S32_ARRAY:
+	case LDMS_V_U64_ARRAY:
+	case LDMS_V_S64_ARRAY:
+	case LDMS_V_F32_ARRAY:
+	case LDMS_V_D64_ARRAY:
+		memcpy(dst, src, ldms_metric_value_size_get(type, count));
```

Comment:
@narategithub it seems like we make no assurances that dst, and src are the same type. Which means that we could be copying garbage. The functions that call this, however, have this information.

Reply:
I'm confused. It looks to me like all of the call sites use `assign_value()` to copy `src` to `dst` of the same type. Are you suggesting that `assign_value()` should have `src_type` and `dst_type` parameters instead of just `type`?

Reply:
@narategithub, this is __in__ assign_value(). I'm suggesting that assign_value() may be broken.